bin/gaffer

```diff
@@ -289,11 +289,8 @@ if [[ -n $DELIGHT ]] ; then
 
 	appendToPath "$DELIGHT/bin" PATH
 	appendToPath "$DELIGHT/python" PYTHONPATH
-	appendToPath "$DELIGHT/shaders" DL_SHADERS_PATH
 
-	appendToPath "$DELIGHT" OSL_SHADER_PATHS
-
-	appendToPath "$GAFFER_ROOT/renderMan/displayDrivers" DL_RESOURCES_PATH
+	appendToPath "$DELIGHT/osl" OSL_SHADER_PATHS
```

Comment:
Right, that's definitely a bad thing to do. I was trying to make the 3Delight shader names a little more convenient by not requiring the `osl/` prefix they need if we only add `$DELIGHT` to `OSL_SHADER_PATHS`. I think that might still be worth doing? But I'm not as sure as I was when I first made this change.

Possible solutions :
1. Add `$DELIGHT:$DELIGHT/osl` to `OSL_SHADER_PATHS`. That would allow existing scenes to still load and also load shaders without the `osl/` prefix. It also means we end up with all of the 3Delight shaders from the `osl` directory in OSL->Shader->Other menu. Which we could hide with `hideShaders()`.
2. Just stick with `OSL_SHADER_PATHS=$DELIGHT` and live with requiring the `osl/` prefix.

I lean towards solution 1 because the `osl/` prefix will be universal, so why require users to always input it. Besides the `hideShaders()` need it comes with, are there other drawbacks to searching both locations?

Reply:
Solution 1 sounds good to me. I suggest commenting the `$DELIGHT` part to say that it's for backwards compatibility, then we can remove it at some point in the future.

The only other possibility I can see would be to only add `$DELIGHT/osl` and then to monkey-patch `OSLShader.loadShader()` to strip any necessary`osl/` prefixes from old files. Maybe we do that when we remove the backwards compatibility though?

Reply:
Sounds good, I have that done in https://github.com/GafferHQ/gaffer/pull/5459/commits/2ff04ae3607033a8c0b756e1b9a314d64f2bea65